Scrapper-Service/plugins/guide2research.py

```diff
@@ -0,0 +1,354 @@
+        links = []
+        for table in conf_tables:
+            try:
+                anchor = table.h4.a
+                if not anchor:
+                    pass
+                else:
+                    link = f"http://www.guide2research.com{anchor['href']}"
+                    links.append(link)
+            except Exception as e:
+                PageParsingError(
+                    f"The following error occured while trying to parse an anchor element {e}")
+        return links
+
+    def _parse_top_conference(self, link: str) -> Conference:
+        """
+        Parses individual top conference page and,
+        return Conference object
+
+        Args:
+        ---
+        link: str
+
+        Returns:
+        ---
+        Conference: object
+        """
+        page = requests.get(link, allow_redirects=True)
+        if not page.status_code == 200:
+            self.logger.error(f"loading {link} failed with {page.status_code}")
+        try:
+            content = page.content
+        except Exception as e:
+            PageParsingError(
+                f"The following error occured while parsing {link} Trace:{e}")
+        soup = bs(content, "html5lib")
+        post_div = soup.find("div", attrs={"class": "single_post"})
+        post_tables = post_div.find_all("table")
+        title = soup.h1.text
+        conf_info = self._get_top_conf_info(table=post_tables[0])
+        rating_info = self._get_top_conf_ranking(name=title, table=post_tables[1])
+        bulk_text = self._get_top_conf_bulk_text(soup)
+        url = conf_info.get("link")
+        deadline = conf_info.get("deadline")
+        metadata = Metadata(__name__,
+                            dt.now(),
+                            link,
+                            self.base_address,
+                            'Guide2Research',
+                            )
+        additional_data = {}
+        additional_data["dates"] = conf_info.get("dates")
+        additional_data["rankings"] = rating_info
+        additional_data["address"] = conf_info.get("address")
+        additional_data["bulkText"] = bulk_text
+        self.logger.info(f"{title} is now added to the database")
+        return Conference(title=title,
+                          url=url,
+                          deadline=deadline,
+                          metadata=metadata,
+                          **additional_data)
+
+    def _get_top_conf_info(self, table: bs) -> Dict[str, object]:
+        """
+        Parses conference info table and
+        returns information dictionary
+
+        Args:
+        ---
+        table: bs4 table
+
+        Returns:
+        ---
+        conference information dictionary
+        """
+        try:
+            tds = table.findAll("td")
+            deadline = tds[1].text.strip()
+            dates = tds[4].text.strip().split("-")
+            address = tds[6].text.strip()
+            link = tds[8].text.strip()
+            return ({"deadline": deadline,
```

Comment:
* deadline should be a date type , use the self.get_date in scrapper class to convert it. Also make sure that you check the format. You may need to pass a new format to parse string. Inspect the method to understand how it works.

Reply:
Done